### ISSUE TITLE
Replace Events with Emitter

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/plinko-probability#49]
Sorry to add this on after the code review, but I noticed this while investigating https://github.com/phetsims/plinko-probability/issues/40.

Events.js has been marked as deprecated and should be replaced with Emitter.js.  Rather than extend Emitter, Ball and Histogram should extend something else, and an Emitter should be create for each type of event that needs to be emitted.

For example, you could create an ```exitedEmitter```, and then replace calls to ```trigger0``` with ```exitedEmitter.emit()```.

expression-exchange has some nice emitter examples if you wish to look at them.

**Comment by veillette** [comment]
I had a look at expression-exchange. Ball and histogram are now extending Object rather than Events. There are no more Events in the sims.. We have now four different type of emitters.  

For the balls, I added a remove listener  (following a pattern I observed in  expression exchange)  to emitter that were attached to balls, to prevent any memory leaks.

 I also commented on why this procedure is unnecessary on the histogramNode.
